coding/transform.go

```diff
@@ -5,26 +5,31 @@ import (
 
 	mc "github.com/jbenet/go-multicodec"
 	mcjson "github.com/jbenet/go-multicodec/json"
+	mccbor "github.com/jbenet/go-multicodec/cbor"
 	ipld "github.com/ipfs/go-ipld"
 )
 
-type codec struct {
+type transformCodec struct {
 	mc.Multicodec
 }
 
-type decoder struct {
+type transformDecoder struct {
 	mc.Decoder
 }
 
-func jsonMulticodec() mc.Multicodec {
-	return &codec{mcjson.Multicodec(false)}
+func JsonMulticodec() mc.Multicodec {
+	return &transformCodec{mcjson.Multicodec(false)}
 }
 
-func (c *codec) Decoder(r io.Reader) mc.Decoder {
-	return &decoder{ c.Multicodec.Decoder(r) }
+func CborMulticodec() mc.Multicodec {
+	return &transformCodec{mccbor.Multicodec()}
```

Comment:
This is exactly that.

What I would love is to replace the parsing of the CBOR once to a data structure always in memory with a parsing that happend on demand and that produces a flow of tokens. Basically, implementing the interface described in #17